### ISSUE TITLE
[Bug]  Excessive Session Creation During SSR with File Session Driver Causing Performance Degradation

### ISSUE THREAD

**Issue by carlosvaldesweb** [manchenkoff/nuxt-auth-sanctum#163]
We are encountering a significant performance issue with the nuxt-sanctum package where multiple sessions are being created during server-side rendering (SSR) in our Nuxt application. It appears that each server-side request generates a new session file in Laravel’s session storage directory (/storage/framework/sessions). This issue is particularly problematic when using the file session driver.

For instance, if a page makes three different API calls, three separate session files are created. This has led to an excessive accumulation of session files, with approximately 800,000 files currently in storage. This excessive number of files is severely impacting performance, particularly when Laravel’s garbage collection process runs, causing requests to become very slow.

This issue seems to occur only during SSR and does not affect client-side rendering. The performance degradation due to the large number of session files affects the overall application responsiveness.

Steps to Reproduce:

- Set up and use the nuxt-sanctum package for handling authentication with Sanctum in a Nuxt application.
- Perform server-side rendering of a page that makes multiple API calls to different endpoints, in my case i have a plugin that makes api call to globals end point, and the user enpoint from package. Also movel between pages that have another api calls.
- Observe that each request generates a new session file in Laravel’s session storage directory.
Expected Behavior:

Sessions should be managed efficiently such that multiple requests during SSR do not lead to the creation of excessive session files. Sessions should remain active and not create new files unnecessarily.

Actual Behavior:

During SSR, each server-side request creates a new session file, resulting in a large number of session files in the storage directory. For example, if a page makes three API calls, three new session files are generated. This results in approximately 800,000 session files in storage, which significantly impacts performance, especially during garbage collection, causing requests to become very slow.

These are session files created in my localhost, but is the same in production
First, when we loads the page i make 3 api calls, to user, globals and help-center, and these 3 files are created
```
a:3:{s:6:"_token";s:40:"fqvDAB707m3gfKvpi0gJrwiVPWid3RPMOIaDwbd5";s:9:"_previous";a:1:{s:3:"url";s:36:"http://localhost:8000/v1/help-center";}s:6:"_flash";a:2:{s:3:"old";a:0:{}s:3:"new";a:0:{}}}
```

```
a:3:{s:6:"_token";s:40:"AZ9GrUfDXH2Gmq3LpjqKBCNF2oE5vFiyVAIb1GEM";s:9:"_previous";a:1:{s:3:"url";s:32:"http://localhost:8000/v1/globals";}s:6:"_flash";a:2:{s:3:"old";a:0:{}s:3:"new";a:0:{}}}
```

```
a:3:{s:6:"_token";s:40:"MuRqLSW8igB8ox0KMi2vfkxcgmi0cYuwEpvviWCO";s:9:"_previous";a:1:{s:3:"url";s:29:"http://localhost:8000/v1/user";}s:6:"_flash";a:2:{s:3:"old";a:0:{}s:3:"new";a:0:{}}}
```

After i've visited other pages, like blog, etc and another files are created
```
a:5:{s:6:"_token";s:40:"crP434MJJEK4L3v5OaugTx3OLvmx7LYeRctbGKud";s:9:"_previous";a:1:{s:3:"url";s:49:"http://localhost:8000/v1/blogs?page=1&per_page=10";}s:6:"_flash";a:2:{s:3:"old";a:0:{}s:3:"new";a:0:{}}s:50:"login_web_59ba36addc2b2f9401580f014c7f58ea4e30989d";i:1;s:17:"password_hash_web";s:60:"$2y$10$OLhYRD/0zOelHxokN8qimu2UJXaQxDQquRmO4y4fwIUi8uR/fkYCi";}
```

**Comment by manchenkoff** [comment]
Hey @carlosvaldesweb, I remember I faced the same issue but the problem turned out to be on the backend side. Can you ensure that even during regular laravel calls via browser it keeps the session in the cookies and does not create a new one after each page refresh?

**Comment by carlosvaldesweb** [comment]
Thank you very much for your response @manchenkoff 

When you mention "regular Laravel calls via browser," are you referring to visiting the API URL directly in the browser?

Just to clarify, the session files are being created from the frontend only when using SSR. This happens both on the first request and on subsequent ones, but only when SSR is involved. No session files are created during client-side rendering.

Thanks again for your help!

**Comment by manchenkoff** [comment]
> When you mention "regular Laravel calls via browser," are you referring to visiting the API URL directly in the browser?

Yes, for example, a health `/up` endpoint or any other route that uses sessions. 

> Just to clarify, the session files are being created from the frontend only when using SSR. This happens both on the first request and on subsequent ones, but only when SSR is involved. No session files are created during client-side rendering.

Laravel recreates the session only when the session cookie is missing from the client side, but since we pass all the cookies from the client side to Nuxt and then to the API, it should work the same way as in client mode. I will double-check that as well, but would be nice to exclude backend misconfiguration first.

**Comment by manchenkoff** [comment]
Hey @carlosvaldesweb, I assume I found the root cause, SSR requests didn't use the same User-Agent as CSR requests so Laravel regenerates the session id during en/decryption process. I published the new 0.4.13 version with the fix, please try it out and let me know if the issue persists!
